test/shared/src/main/scala-2.x/zio/test/mock/MockableMacro.scala

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2019-2020 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.test.mock
+
+import scala.reflect.macros.whitebox.Context
+
+import com.github.ghik.silencer.silent
+
+/**
+ * Generates method tags for a service into annotated trait/object.
+ */
+private[mock] object MockableMacro {
```

Comment:
I wonder if either the macro functionality or the mocking functionality should be its own project.

Reply:
The macros are closly tried to the implementation details and whenever we change ZIO Mocks, the Mockable macro must be updated aswell, that is why I've moved this into `zio/zio`. We could extract this into seperate subproject though. Since most of `zio-macros` (delegate family) got obsoleted with `ZLayer` and `Has` we are planning to move also `@accessible` into `zio/zio` (maintaining a seperate repository for just one macro feels like too much overhead).

Reply:
Yes, if we do anything agree it would be in the same repository. Would be like the ZIO Test Magnolia project. That would solve the problem of updating things at the same time while potentially being more modular.